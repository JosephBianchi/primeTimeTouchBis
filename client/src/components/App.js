import React, { useState } from 'react';

import styles from './App.module.css';
import { math } from '../apis/math.js';
import Form from './FormComponent';
import PrimeNumbers from './PrimeNumbersComponent';
import Header from './HeaderComponent';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { MdCheckCircle } from "react-icons/md";
import { MdSentimentDissatisfied } from "react-icons/md";
import loading from '../assets/ballLoader.gif';

const App = (props) => {
  const { propsPrimeNumbers, propsMaxInt, propsErrorMessage, propsIsLoading, propsLastSearch } = props;
  const [primeNumbers, setPrimeNumbers] = useState(propsPrimeNumbers || {});
  const [maxInt, setMaxInt] = useState(propsMaxInt || '');
  const [lastSearch, setLastSearch] = useState(propsLastSearch || '');
  const [errorMessage, setErrorMessage] = useState(propsErrorMessage || '');
  const [isLoading, setIsLoading] = useState(propsIsLoading || false);

  const handleSubmit = () => {
    setLastSearch(maxInt);
    setIsLoading(true)
    fetch(`${math}/prime`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({maxInt: parseInt(maxInt)})
    })
    .then((response) => {
      return response.json();
    })
    .then((data) => {
      setPrimeNumbers(data);
      setMaxInt('')
      console.log(data.errorMessage);
      setErrorMessage(data.errorMessage)
      setIsLoading(false);
      console.log(data);
      return data;
    })
    .catch((err) => {
        console.log("Something went wrong!", err);
    });
  }

  const changeHandler = (event) => {
    setMaxInt(event.target.value)
  }

  return (
    <div className={styles.App}>
      <Header />
      <Container fluid>
        <Row>
          <Col md={{ span: 4, offset: 2 }}>
            <h3>Median Prime Numbers
              <strong>
              {
                (lastSearch && !errorMessage) ?
                  ` for non-inclusive max of: ${lastSearch}`:
                null
              }
              </strong>
            </h3>  
            <Form
              label='Non-Inclusive Max'
              onSubmit={ handleSubmit }
              value={ maxInt }
              onChange={ changeHandler }
              placeholder='Enter a Number greater than or equal to 3'
            />
          </Col>
        </Row>
        <Row>
          {
            isLoading ? <img className={styles.loading} src={loading} alt="loading" /> :
            <>
              <Col md={{ span: 4, offset: 2 }}>
                <div>
                  {
                    (errorMessage.length === 0 && primeNumbers.median) ?
                    <PrimeNumbers
                      median={ primeNumbers.median }
                    /> : (errorMessage.length > 0) ?
                    <p className={ styles.errorMessage }>{ errorMessage }</p> :
                      null
                  }
                </div>
              </Col>
              <Col>
                {
                  (errorMessage.length === 0 && primeNumbers.median) ?
                    <MdCheckCircle className={ styles.successIcon } /> :
                    (errorMessage.length > 0) ?
                      <MdSentimentDissatisfied className={ styles.warningIcon } /> :
                      null
                }
              </Col>
            </>
          }
        </Row>
      </Container>
    </div>
  );
}

export default App;
