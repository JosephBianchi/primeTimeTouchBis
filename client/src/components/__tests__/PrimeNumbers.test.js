import React from 'react';
import { mount } from 'enzyme';

import App from '../App';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <App propsPrimeNumbers={{
      median: [ 3, 2 ],
      primes: [ 2, 3 ],
      errorMessage: ''
    }} />
  );
})

it('should render a list of median numbers that match the list length of the propsPrimeNumbers array', () => {
  expect(wrapped.find('.card').length).toEqual(2);
})

it('should match the snapshot', () => {
  expect(wrapped.html()).toMatchSnapshot();
});
