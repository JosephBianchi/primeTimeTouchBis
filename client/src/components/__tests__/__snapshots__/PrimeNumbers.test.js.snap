// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `"<div class=\\"App\\"><div class=\\"header jumbotron jumbotron-fluid\\"><div class=\\"container\\"><h1>Prime Time</h1><h5>By Joseph Bianchi</h5><p>This is an application that was built with<strong> React Hooks</strong>, the application communicates with an<strong> Express Server</strong>. The application accepts a 'max number' and returns the median prime number from a list of all the prime numbers that are less than the provided 'max number'.</p><small>Ancient Alogrithmic Approach: Sieve of Eratosthenes</small></div></div><div class=\\"container-fluid\\"><div class=\\"row\\"><div class=\\"col-md-4 offset-md-2\\"><h3>Median Prime Numbers<strong></strong></h3><form class=\\"\\"><div class=\\"form-group\\"><label class=\\"form-label\\">Non-Inclusive Max</label><input placeholder=\\"Enter a Number greater than or equal to 3\\" type=\\"number\\" class=\\"form-control form-control-lg\\" value=\\"\\"><button type=\\"submit\\" class=\\"formButton btn btn-primary\\">Submit</button></div></form></div></div><div class=\\"row\\"><div class=\\"col-md-4 offset-md-2\\"><div><div><div class=\\"numCard card\\"><p>3</p></div><div class=\\"numCard card\\"><p>2</p></div></div></div></div><div class=\\"col\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" class=\\"successIcon\\" height=\\"1em\\" width=\\"1em\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\\"></path></svg></div></div></div></div>"`;
